### DISCUSSION TITLE
Word-wrap when adding text to pdf using pdfpig C#

### DISCUSSION THREAD

**Discussion by T0nyKo** [UglyToad/PdfPig#934] [Q&A]
Hi, is there any functionality in pdfpig that i can use to make this string wrap around to the next line on this pdf with margin?
Is there anyway I can use a pdfrectangle instead of the pdfpoint to show how much of a rectangle the text should fill out?

Any help would be greatly appreciated. 

This is my C# code: 

```
var page = builder.AddPage(UglyToad.PdfPig.Content.PageSize.A4, true);

var imgX = 70; // Convert.ToInt32(ConfigurationManager.AppSettings["imgX"]);//cmToPdfUnits(2.5);   //70.867
var imgY = 680; // Convert.ToInt32(ConfigurationManager.AppSettings["imgY"]);//cmToPdfUnits(14);    //396.85    
var imgWidth = 523; // Convert.ToInt32(ConfigurationManager.AppSettings["imgWidth"]);//cmToPdfUnits(16);    //453.54
var imgHeight = 780; // Convert.ToInt32(ConfigurationManager.AppSettings["imgHeight"]);//cmToPdfUnits(12);   //340.16
page.AddJpeg(imgstream, new PdfRectangle(imgX, imgY, imgWidth, imgHeight));

PdfDocumentBuilder.AddedFont font = builder.AddStandard14Font(Standard14Font.TimesRoman);
string displaytext = "Important Update Regarding the Workday Deployment";
page.AddText(displaytext, 24, new PdfPoint(70, 550), font);
```

![image](https://github.com/user-attachments/assets/cc019147-a869-4c1b-a1d1-dded16aff496)

**Comment by EliotJones** [comment]
Word wrap is a thorny problem that is somewhat out of scope for this library, see the problems with greedy wrap versus hard/soft wrap and dynamic programming approaches for details. We'd welcome contributions to support it though.
